examples/bigcommerce/lib/makeswift/context.tsx

```diff
@@ -0,0 +1,149 @@
+import { createContext, ReactNode, useCallback, useContext, useEffect, useState } from 'react'
+
+export type Image = {
+  url: string
+  alt: string
+}
+export type Product = {
+  url: string
+  id: number
+  name: string
+  description: string
+  price: string
+  defaultImage: Image
+  images: Image[]
+  categories: Category[]
+}
+export type Category = 'Shop All' | 'Plants' | 'Pots'
```

Comment:
When using enum types like these I would suggest we actually define a value so that we're not hand-writing string values elsewhere. i.e.,

```ts
const ProductCategory = {
  ShopAll: 'shop-all',
  Plants: 'plants',
  Pots: 'pots',
} as const

type ProductCategory = typeof ProductCategory[keyof typeof ProductCategory]

// Use it
const category: Category = ProductCategory.Plants
```